packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -528,6 +534,50 @@ export class DefaultCrudRepository<
     return includeRelatedModels<T, Relations>(this, entities, include, options);
   }
 
+  /** Converts an entity object to a JSON object to check if it contains navigational property.
+   * Throws an error if `entity` contains navigational property.
+   * Also removed id property  for replaceById operation (mongo use case).
+   *
+   * @param entity
+   * @param options when attributw replacement is set to true, delete the id property to operate
+   * replaceById method.
+   */
+  protected ensurePersistable<R extends T>(
+    entity: R | DataObject<R>,
+    options: {replacement?: boolean; idProperty?: string} = {},
+  ): legacy.ModelData<legacy.PersistedModel> {
+    // FIXME(bajtos) Ideally, we should call toJSON() to convert R to data object
+    // Unfortunately that breaks replaceById for MongoDB connector, where we
+    // would call replaceId with id *argument* set to ObjectID value but
+    // id *property* set to string value.
+    /*
+    const data: AnyObject =
+      typeof entity.toJSON === 'function' ? entity.toJSON() : {...entity};
+    */
+
+    // proposal solution from agnes: delete the id property in the target dtat
+    // when runs replaceById
+
+    const data: AnyObject = new this.entityClass(entity);
+
+    const def = this.entityClass.definition;
+    for (const r in def.relations) {
```

Comment:
The way how we detect the navigational property is reasonable to me 👍

Reply:
Improvement to consider: not the scope of this PR, if the same check applies to all CRUD functions, maybe we should define it as an interceptor?